### ISSUE TITLE
Bt works , but didn't connect to my phone

### ISSUE THREAD

**Issue by ItsVixano** [AsteroidOS/meta-mooneye-hybris#1]
hi , i'm using a mooneye , and after installed it , i tried to connect to my phone but , he make a loop of connect/disconnect 
i love this project and i want to use it every day , but i cannot do this
it see be a phone issue "Whyred with EvoX" but i don't know 

i'm using the last snapshot release

sorry for my grammar , i'm italian

**Comment by Qiangong2** [comment]
Can anyone confirm that this is still the issue with mooneye? Or does bluetooth now work?

**Comment by saladhax** [comment]
I just flashed it to my Ticwatch E, and I can confirm this is still an issue. Just keeps bouncing back and forth between connect and disconnect.

**Comment by sayopaul** [comment]
thanks for the info. I am seriously considering flashing to my Ticwatch E however I want to wait till bluetooth is working

**Comment by AyoPrez** [comment]
Yeah, I just flash my Ticwatch E and this is still happening. Does someone know where or what is the problem?

**Comment by nuukedo29** [comment]
Same issue here, i would've got the logcat, but adb seems to be dead aswell

**Comment by archaicmeander** [comment]
Still an issue on Ticwatch E with AsteroidOS 1.1-nightly. Infinite connect-disconnect loop. Anyone have any suggestions? Has any headway been made?

**Comment by Sustenance** [comment]
@FlorentRevest I know it's been a couple years, but I see that you did spend time [backporting and patching the BT](https://github.com/AsteroidOS/meta-mooneye-hybris/blob/master/recipes-kernel/linux/linux-mooneye/0004-Backport-HCI_STP-from-Mediatek.patch) for mooneye. 
I tried watching `/dev/kmsg` but it was hard for me to pinpoint what might be going wrong (I'm also not experienced with debugging drivers). I can provide that log though, if you have time to look. If not, do you have any pointers for where one of us should start investigating? Thanks in advance.

**Comment by FlorentRevest** [comment]
Hey @Sustenance, thanks for reaching out :)
If I recall correctly, MediaTek android kernels come with a bluetooth driver that does not support HCI and even though it's fine on Android, it makes it impossible to use a BlueZ stack. I had found a MTK HCI driver in some random Ubuntu Touch kernel. It works on harmony and it seems to kinda work (a little bit) on mooneye but I don't remember exactly what stopped working. Maybe you could use `hcidump` to get HCI logs of what happens during a connection to your watch and this could be compared with another Asteroid watch where the connection works. I unfortunately no longer have the time necessary to look into this.

**Comment by KulguninAr4i** [comment]
> Apparently there is also https://github.com/mer-hybris/bluebinder that creates a virtual HCI devices and bridges data and events to the Android binder. This is available for Android bases 8 and up. We currently have two devices that are based on Android 8: mooneye and sawfish (Currently on Nougat but I have it working on Oreo too).
> 
> I don't own a mooneye but I can try this in the future on my sawfish. Keep in mind that this a low priority thing for me as Bluetooth works fine on sawfish using the traditional patches (backport 4.1 patches and patch Bluesleep).

Hello. I am actively testing this watch. Has there been any progress for the Ticwatch S / E?

**Comment by MagneFire** [comment]
> > Apparently there is also https://github.com/mer-hybris/bluebinder that creates a virtual HCI devices and bridges data and events to the Android binder. This is available for Android bases 8 and up. We currently have two devices that are based on Android 8: mooneye and sawfish (Currently on Nougat but I have it working on Oreo too).
> > I don't own a mooneye but I can try this in the future on my sawfish. Keep in mind that this a low priority thing for me as Bluetooth works fine on sawfish using the traditional patches (backport 4.1 patches and patch Bluesleep).
> 
> Hello. I am actively testing this watch. Has there been any progress for the Ticwatch S / E?

Unfortunately not for the moment. I didn't have time to work on this. When I will get time for it I will pretty much just use the flow described in https://sailfishos.org/wiki/Architecture (i.e. Bluez5 <> kernel VHCI <> bluebinder <> libgbinder <> Android BSP HAL: android.hardware.bluetooth )
Once I have something I will of course update this thread :smile:
